builder/builder.go

```diff
@@ -524,25 +532,56 @@ func (b *Builder) BuildChroots(template *x509.Certificate, privkey *rsa.PrivateK
 
 // BuildUpdate will produce an update consumable by the swupd client
 func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string, skipSigning bool, publish bool, keepChroots bool) error {
+	var timer stopWatch
+	defer timer.WriteSummary(os.Stdout)
+
 	if format != "" {
 		b.Format = format
 	}
 
+	var err error
+	// TODO: move this to parsing configuration time.
+	var mixVerUint uint32
+	mixVerUint, err = parseUint32(b.Mixver)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't parse mix version")
+	}
+
 	// Ensure the format dir exists.
 	formatDir := filepath.Join(b.Statedir, "www", "version", "format"+b.Format)
-	err := os.MkdirAll(formatDir, 0777)
+	err = os.MkdirAll(formatDir, 0777)
 	if err != nil {
 		return errors.Wrapf(err, "couldn't create the format directory")
 	}
 
 	// Create update metadata for the mix.
-	updatecmd := exec.Command(prefixflag+"swupd_create_update", "-S", b.Statedir, "--minversion", strconv.Itoa(minVersion), "-F", b.Format, "--osversion", b.Mixver)
-	updatecmd.Stdout = os.Stdout
-	updatecmd.Stderr = os.Stderr
-	err = updatecmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to create update metadata")
+	timer.Start("create update")
+	if UseNewSwupdServer {
+		// TODO: move this to parsing configuration / parameter time.
+		if minVersion < 0 || minVersion > math.MaxUint32 {
+			return errors.Errorf("minVersion %d is out of range", minVersion)
+		}
+		// TODO: support format strings in CreateManifests. See
```

Comment:
We agree this isn't necessary, correct?

Reply:
Correct. I haven't pushed the updated version yet.